libgc/include/gc.h

```diff
@@ -835,6 +835,10 @@ typedef GC_PTR (*GC_fn_type) GC_PROTO((GC_PTR client_data));
 GC_API GC_PTR GC_call_with_alloc_lock
         	GC_PROTO((GC_fn_type fn, GC_PTR client_data));
 
+GC_API void GC_start_blocking GC_PROTO((void));
```

Comment:
attn @joncham

Reply:
Note that upstream Boehm has similar concept, but is transitioned via a callback function both in and out rather than setting a boolean. This is less flexible, but maybe there is reason they considered this design (stack/registers at transition frames?)

```
/* Wrapper for functions that are likely to block (or, at least, do not */
/* allocate garbage collected memory and/or manipulate pointers to the  */
/* garbage collected heap) for an appreciable length of time.  While fn */
/* is running, the collector is said to be in the "inactive" state for  */
/* the current thread (this means that the thread is not suspended and  */
/* the thread's stack frames "belonging" to the functions in the        */
/* "inactive" state are not scanned during garbage collections).  It is */
/* allowed for fn to call GC_call_with_gc_active() (even recursively),  */
/* thus temporarily toggling the collector's state back to "active".    */
GC_API void * GC_CALL GC_do_blocking(GC_fn_type /* fn */,
                                void * /* client_data */) GC_ATTR_NONNULL(1);
```

https://github.com/ivmai/bdwgc/blob/9468065bc9e5955c930e7b5e6e00c6b1b77fa468/include/gc.h#L1502

Reply:
Okay, I think it's difference in design after reading this PR more. Boehm allows you to transition to "no GC" state with GC memory in frames above current one. They produce an interleaved stack to be scanned, with GC and non-GC reasons. I believe for something like calling a system call from icall/pinvoke. 

This design, correct me if wrong, is about worker threads which promise to have no GC memory on any stack frames when disabled?

Reply:
Yes, exactly right. This design is intentionally "simpler" than the upstream one.

I should probably document that in a comment somewhere.

Reply:
Added cb31f6bb035bf7423e392b02b3287ae1201c8614.